locations/jclouds/src/main/java/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -596,21 +599,35 @@ protected JcloudsSshMachineLocation obtainOnce(ConfigBag setup) throws NoMachine
                     userCredentials = initTemplateForCreateUser(template, setup);
                 }
 
-                //FIXME initialCredentials = initUserTemplateOptions(template, setup);
-                for (JcloudsLocationCustomizer customizer : getCustomizers(setup)) {
-                    customizer.customize(this, computeService, template);
-                    customizer.customize(this, computeService, template.getOptions());
+                templateTimestamp = Duration.of(provisioningStopwatch);
+                // "Name" metadata seems to set the display name; at least in AWS
+                // TODO it would be nice if this salt comes from the location's ID (but we don't know that yet as the ssh machine location isn't created yet)
+                // TODO in softlayer we want to control the suffix of the hostname which is 3 random hex digits
+                template.getOptions().getUserMetadata().put("Name", cloudMachineNamer.generateNewMachineUniqueNameFromGroupId(setup, groupId));
```

Comment:
looks like the `template.getOptions()` offers also `nodeNames` which is used for `aws` api to assign the `name` to the NodeMetadata. so maybe this also answer the comment 

// "Name" metadata seems to set the display name; at least in AWS